.github/workflows/terraform-cicd.yaml

```diff
@@ -0,0 +1,63 @@
+name: 'Terraform-Workflow'
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    types: [opened, edited, reopened, ready_for_review, review_requested]
```

Comment:
@sjoshi-jpl do we want to run these TF workflows on PRs/branches? I was thinking we would only want to deploy these things when they are merged to main?

Reply:
@jordanpadams yes but I can limit the number of times this workflow gets executed by only running `terraform plan` on PR creation and push to main.

[more details](https://github.com/NASA-PDS/nucleus/pull/56#pullrequestreview-1522517881)